compiler/AST/expr.cpp

```diff
@@ -247,7 +247,7 @@ bool Expr::isStmtExpr() const {
     retval = (parent->indexGet() != this && parent->iteratorGet() != this) ? true : false;
 
   } else {
-    retval = isBlockStmt(parentExpr);
+    retval = isDirectlyUnderBlockStmt(this);
```

Comment:
This seems pretty unfortunate.  Do you expect that it will be rare to need isDirectlyUnderBlockStmt(e) instead of isBlockStmt(e->parentExpr)?